hpke/aead.go

```diff
@@ -67,6 +164,47 @@ func (c *sealCtx) Seal(pt, aad []byte) ([]byte, error) {
 	return ct, nil
 }
 
+// Marshal serializes an HPKE sealer according to the format specified below.
+// (expressed in TLS syntax). Note that this format is not defined by the HPKE
+// standard.
+//
+// enum { sealer(0), opener(1) } HpkeRole;
+//
+// struct {
+//     HpkeKemId kem_id;   // draft-irtf-cfrg-hpke-06
+//     HpkeKdfId kdf_id;   // draft-irtf-cfrg-hpke-06
+//     HpkeAeadId aead_id; // draft-irtf-cfrg-hpke-06
+//     opaque exporter_secret<0..255>;
+//     opaque key<0..255>;
+//     opaque base_nonce<0..255>;
+//     opaque seq<0..255>;
+// } HpkeContext;
+//
+// struct {
+//   HpkeRole role = 0; // sealer
+//   HpkeContext context;
+// } HpkeSealer;
+func (c *sealCtx) Marshal() ([]byte, error) {
+	rawCtx, err := c.encdecCtx.marshal()
+	if err != nil {
+		return nil, err
+	}
+	c.raw = append([]byte{0}, rawCtx...)
```

Comment:
Here, you are caching the `raw` value, but not using it in a subsequent call to Marshal.

Also, you can still use the `raw` variable inside the `encdecCtx` object. Right now, you have two `raw` variables.

 However, if you cache at time T, then after doing some encryptions, the serialized value must be updated anyway. So you might not need `raw` variable at all.

Reply:
Removed `raw` from `encdecContext`, `sealContext`, and `openContext`.

I realized caching raw is a bad idea, since Seal() and Open() advance the sequence number.